bug: weird error for accessing tuple type field
#22633
